x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.tsx

```diff
@@ -25,15 +26,22 @@ import { SelectEngineType } from './select_engine_type';
 
 export const EngineCreation: React.FC = () => {
   const { search } = useLocation() as Location;
-  const { method } = parseQueryParams(search);
+  const { method, ...params } = parseQueryParams(search);
 
   const { engineType, currentEngineCreationStep } = useValues(EngineCreationLogic);
-  const { setIngestionMethod } = useActions(EngineCreationLogic);
+  const { setIngestionMethod, setEngineType, setCreationStep, setSelectedIndex } =
+    useActions(EngineCreationLogic);
 
   useEffect(() => {
     if (typeof method === 'string') {
       setIngestionMethod(method);
     }
+    const esIndexParam = params[ESINDEX_QUERY_PARAMETER];
+    if (typeof esIndexParam === 'string') {
+      setEngineType('elasticsearch');
```

Comment:
suggestion(non-blocking): It might be a little neater to have a single `initializeEngineCreation` action that takes care of all this setup itself, either by calling those separate actions in a listener, or by just triggering the reducers.